### ISSUE TITLE
Intiface opens but stalls on Linux

### ISSUE THREAD

**Issue by LeGrosFromage** [intiface/intiface-central#81]
Using Gentoo linux, opening Intiface Central (both 2.4.0 and 2.4.2)  creates the app window, shows the logo and proceeds no further:

![image](https://github.com/intiface/intiface-central/assets/97702146/6619fef5-c737-429d-9d74-1e7d390daa51)

I've deleted ~/.local/share/com.nonpolynomial.intiface_central/ but the problem persists.

The log shows:
```
[I] Global Loggy: Running main builder
[I] Global Loggy: Intiface Central Starting...
[I] Global Loggy: Device configuration file does not exist, returning 0.0.
[I] Global Loggy: {}
[I] Global Loggy: []
[E] Global Loggy: Error loading cubit! Deleting configs and creating new ones.
```
This has happened on all recent versions, so I tend to use Windows for this, but I'd like to get it working on Linux.
Linux specs - Gentoo, X.org, kernel 6.3.8, Ryzen CPU, 48Gb Ram, 500Gb disc free.

FWIW - 2.42 does the same thing on Mac, although 2.4.0 works fine.

**Comment by qdot** [comment]
Ok this is actually 2 separate issues.

Linux:

I am going to bet this is a libssl compat issue. I'm getting this constantly across random linux distros.

I'm gonna try to vendor in openssl, which *should* hopefully fix this.

macOS:

You're running on an intel machine, right? My builds on my m1 don't seem to be as universal as they're supposed to be (I'm guessing my intel builds work because of rosetta, but that's a one-way street), probably at the rust level, so I need to figure out how to make sure we build all of the correct architectures there.

**Comment by qdot** [comment]
Closing this as it's now split into #83 and #85, which will both be fixed in v2.4.4
